artifactory/commands/gradle/gradle.go

```diff
@@ -231,6 +241,52 @@ func (gc *GradleCommand) setResult(result *commandsutils.Result) *GradleCommand
 	return gc
 }
 
+type InitScriptAuthConfig struct {
+	ArtifactoryURL           string
+	ArtifactoryRepositoryKey string
+	ArtifactoryUsername      string
+	ArtifactoryAccessToken   string
+}
+
+// GenerateInitScript generates a Gradle init script with the provided authentication configuration.
+func GenerateInitScript(config InitScriptAuthConfig) (string, error) {
+	tmpl, err := template.New("gradleTemplate").Parse(gradleInitScript)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse Gradle init script template: %s", err)
+	}
+
+	var result strings.Builder
+	err = tmpl.Execute(&result, config)
+	if err != nil {
+		return "", fmt.Errorf("failed to execute Gradle init script template: %s", err)
+	}
+
+	return result.String(), nil
+}
+
+// WriteInitScript writes the Gradle init script to the Gradle user home `init.d` directory,
+// which stores initialization scripts.
+func WriteInitScript(initScript string) error {
+	gradleHome := os.Getenv(UserHomeEnv)
```

Comment:
lets use GetUserHomeDir as it has some special logic for windows and other cases